cmd/collector/main.go

```diff
@@ -80,21 +86,33 @@ func main() {
 				logger.Fatal("Failed to create span writer", zap.Error(err))
 			}
 
+			requireLockAndSamplingStore, err := strategyStoreFactory.RequiresLockAndSamplingStore()
+			if err != nil {
+				logger.Fatal("Failed to determine if lock and sampling store is required.", zap.Error(err))
+			}
+			var lock distributedlock.Lock
+			var samplingStore samplingstore.Store
+			if requireLockAndSamplingStore {
```

Comment:
What is the purpose of this check? I see that all factories implement the Create method (but many return nils), isn't that the same signal as the extra bool?

Reply:
Yeah, I was torn on this. I added it b/c I liked the explicitness of only attempting to create a sampling store/lock if it was necessary. Without this adaptive sampling will log something like:

"lock and sampling store nil. this is probably b/c your backend does not support adaptive sampling"

However, after all the factory/interface/implementation changes I question its value. Let me take a pass at removing it.

Reply:
I think another aspect here is that originally the storage factory was not meant to be this all encompassing central entry point, i.e. in theory you should be able to configure Cassandra-based Span storage and another backend for Dependencies or Sampling. This is why the env var is specifically SPAN_STORAGE_TYPE, but we never really made clear separation.

Reply:
I have removed the "requireLockAndSamplingStore" functionality and feel like things look cleaner. It will be easy to revert if we want to go back.

> I think another aspect here is that originally the storage factory was not meant to be this all encompassing central entry point, i.e. in theory you should be able to configure Cassandra-based Span storage and another backend for Dependencies or Sampling. This is why the env var is specifically SPAN_STORAGE_TYPE, but we never really made clear separation.

This would require some work in `plugin/storage/factory.go` but may not be that bad. I think most of the code would be isolated here.